test/unit/test_utils.py

```diff
@@ -21,3 +25,67 @@ def test_get_schema():
     schema = local_sdi.get_schema(metadata_out["provides"]["oidc-client"]["schema"])
 
     assert schema == metadata_out["provides"]["oidc-client"]["schema"]
+
+
+PROXY_URLS = {i: f"http://a:800{str(i)}" for i in range(6)}
+
+
+@pytest.mark.parametrize(
+    "env_dict, expected_proxies",
+    [
+        # Null test
+        ({}, {"http": None, "https": None, "no-proxy": None}),
+        # Typical proxy args
+        (
+            {
+                "HTTP_PROXY": PROXY_URLS[0],
+                "HTTPS_PROXY": PROXY_URLS[1],
+                "NO_PROXY": PROXY_URLS[2],
+            },
+            {
+                "http": PROXY_URLS[0],
+                "https": PROXY_URLS[1],
+                "no-proxy": PROXY_URLS[2],
+            },
+        ),
+        # Juju proxy args
+        (
+            {
+                "JUJU_CHARM_HTTP_PROXY": PROXY_URLS[0],
+                "JUJU_CHARM_HTTPS_PROXY": PROXY_URLS[1],
+                "JUJU_CHARM_NO_PROXY": PROXY_URLS[2],
+            },
+            {
+                "http": PROXY_URLS[0],
+                "https": PROXY_URLS[1],
+                "no-proxy": PROXY_URLS[2],
+            },
+        ),
+        # Ensure Juju take priority over regular proxy
+        (
+            {
+                "JUJU_CHARM_HTTP_PROXY": PROXY_URLS[0],
+                "JUJU_CHARM_HTTPS_PROXY": PROXY_URLS[1],
+                "JUJU_CHARM_NO_PROXY": PROXY_URLS[2],
+                "HTTP_PROXY": PROXY_URLS[3],
+                "HTTPS_PROXY": PROXY_URLS[4],
+                "NO_PROXY": PROXY_URLS[5],
+            },
+            {
+                "http": PROXY_URLS[0],
+                "https": PROXY_URLS[1],
+                "no-proxy": PROXY_URLS[2],
+            },
+        ),
+    ],
+)
+def test_get_proxy_settings_from_env(env_dict, expected_proxies):
+    # Patch the environment with our proxy settings
+    with mock.patch.dict(os.environ, env_dict, clear=True):
+        proxies = _get_proxy_settings_from_env()
+        assert proxies == expected_proxies
+
+
+# TODO (ca-scribner): Need a test that simulates a proxy to ensure we can get
```

Comment:
I would like to see that test for `_get_schema_response_from_remote`. As you said you should be able to just mock/patch the `requests.get` call and make sure it is called with the correct proxy settings.